app/models/vm_scan.rb

```diff
@@ -242,7 +244,8 @@ def call_snapshot_delete
           #       or, make type-specific Job classes.
           if vm.kind_of?(ManageIQ::Providers::Openstack::CloudManager::Vm)
             vm.ext_management_system.vm_delete_evm_snapshot(vm, mor)
-          elsif vm.kind_of?(ManageIQ::Providers::Microsoft::InfraManager::Vm)
+          elsif vm.kind_of?(ManageIQ::Providers::Microsoft::InfraManager::Vm) ||
+                vm.kind_of?(ManageIQ::Providers::Azure::CloudManager::Vm)
```

Comment:
The condition for creating the snapshot is:
```ruby
vm.kind_of?(ManageIQ::Providers::Azure::CloudManager::Vm) && vm.require_snapshot_for_scan?
```
Shouldn't that be the same condition for deleting the snapshot? The same question applies to the delete snapshot in the about code path.

Reply:
Yep.  Fixing.